File: LetsPlant/app/src/main/java/com/e/letsplant/data/UserViewModel.java
package com.e.letsplant.data;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class UserViewModel extends ViewModel {
    private final MutableLiveData<User> userMutableLiveData = new MutableLiveData<>();

    public void setUserMutableLiveData(User user){
        userMutableLiveData.setValue(user);
    }
    public LiveData<User> getUserMutableLiveData() {
        return userMutableLiveData;
    }
}

File: LetsPlant/app/src/main/java/com/e/letsplant/data/UserViewModel.kt
package com.e.letsplant.data

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class UserViewModel : ViewModel() {
    private val userMutableLiveData = MutableLiveData<User?>()
    fun setUserMutableLiveData(user: User?) {
        userMutableLiveData.value = user
    }

    fun getUserMutableLiveData(): LiveData<User?> {
        return userMutableLiveData
    }
}